crates/bevy_color/src/color_gradient.rs

```diff
@@ -0,0 +1,104 @@
+use crate::Mix;
+use bevy_math::curve::{cores::EvenCoreError, Curve, Interval, SampleCurve};
+
+/// A curve whose samples are defined by a collection of colors.
+#[derive(Clone, Debug)]
```

Comment:
```suggestion
#[derive(Clone, Debug)]
#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
#[cfg_attr(feature = "bevy_reflect", derive(bevy_reflect::Reflect))]
```

This ensures `serde` and `Reflect` compatibility is added when those feature flags are enabled.

Reply:
The `serde` part here doesn't work since we're now using a concrete type parameter for `I` which is neither `Serialize` nor `Deserialize`. I guess this is just surfacing an issue that wasn't visible before. The only way that would probably work is if the we don't use function pointers but some kind of struct which impls `Fn` in this place, right? 🤔

Reply:
That is an interesting point! I think for now just leave it as you have it without the `serde` integration and we can save that for a follow-up PR. I suspect we would implement a custom `Serialize` and `Deserialize` where we choose the function pointer based on the colour type.